pkg/cache/multi_informer.go

```diff
@@ -0,0 +1,64 @@
+package cache
+
+import (
+	"fmt"
+	"time"
+
+	toolscache "k8s.io/client-go/tools/cache"
+)
+
+type multiNamesapceIndexInformer struct {
+	indexInformers map[string]toolscache.SharedIndexInformer
+}
+
+func (m *multiNamesapceIndexInformer) AddIndexers(indexers toolscache.Indexers) error {
+	return nil
+}
+
+func (m *multiNamesapceIndexInformer) GetIndexer() toolscache.Indexer {
+	return nil
+}
+
+func (m *multiNamesapceIndexInformer) AddEventHandler(handler toolscache.ResourceEventHandler) {
+	for namespace, i := range m.indexInformers {
+		fmt.Printf("\nadding event handler: %v for namespace: %v\n", i, namespace)
+		i.AddEventHandler(handler)
+	}
+}
+
+func (m *multiNamesapceIndexInformer) AddEventHandlerWithResyncPeriod(handler toolscache.ResourceEventHandler, resyncPeriod time.Duration) {
+	for _, i := range m.indexInformers {
+		i.AddEventHandlerWithResyncPeriod(handler, resyncPeriod)
+	}
+}
+
+func (m *multiNamesapceIndexInformer) GetStore() toolscache.Store {
```

Comment:
@hasbro17 This is the stuff that would be hard to expose. This, `GetController` and the `LastSyncResourceVersion()` 

I think it may be fine to not have access to the controller or the store. People should very rarely need access to these from what I have seen. (especially the Store). Also have already set up the watches and the like so not having access to the LastSyncResourceVersion or the controller shouldn't cause issues as far as I can tell. 


I think that is the biggest drawback from this approach.